Overview.bs

```diff
@@ -1688,11 +1688,55 @@ guidance that encoder implementations may want to consider.
 
 <b>Utilize an existing font subsetter implementation</b>
 
-<!-- TODO: A high quality encoder should preserve functionality as discussed above. Existing font subsetters can be used to assist
-     constructing encodings that meet this requirement. They provide all of the logic needed to preserve functionality. High quality
-     open source subsetters exist -->
+As discussed in [[#encoder]] a high quality encoder should preserve the functionality of the original font. One way to
+achieve this is to leverage an existing font subsetter implementation to produce font subsets that retain the functionality
+of the original font. The IFT patches can then be derived from these subsets.
+
+A font subsetter produces a [=font subset=] from an input font based on a desired [=font subset definition=]. The practice of reliably
+subsetting a font is well understood and has multiple open-source implementations (a full formal description is
+beyond the scope of this document). It typically involves a reachability analysis, where the data in tables is examined
+relative to the font subset definition to see which portions can be reached by any possible content covered by the subset definition.
+Any reachable data is retained in the generated font subset, while any unreachable data may be removed.
+
+Example pseudo code for a very basic encoder implementation using a font subsetter:
+
+```
+# Encodes a font (full_font) into an incremental font that starts at base_subset_def
+# and can incrementally add any of subset_definitions. Returns the IFT encoded font
+# and set of associated patches.
+encode_as_ift(full_font, base_subset_def, subset_definitions):
+  base_font = subset(full_font, base_subset_def)
+  patches = encode_node(full_font, base_font, base_subset_def, subset_definitions)
+  return base_font, patches
+
+# Update base_font to add all of the ift patch mappings to reach any of
+# subset_definitions and produces the associated patches.
+encode_node(full_font, base_font, cur_def, subset_definitions):
+  patches = []
+  next_fonts = []
+  
+  for each subset_def in subset_definitions not fully covered by cur_def:
+    next_def = subset_def union cur_def
+    next_font = subset(full_font, next_def)
+    let patch_url be a new unique url
+    
+    add patch map entry into base_font from (subset_def - cur_def) to patch_url
+    patches += encode(full_font, next_font, next_def, subset_definitions)
```

Comment:
Is this line left over from an earlier edit? I don't understand how this patches line relates to the one at 1730.

Reply:
This recursive call generates all of the patches for the lower levels of the graph while on line 1730 patches is further extended to add all of the patches associated with the current level. So encode(..., next_font,.. ) will generate any patches from next_font to anything else reachable from it, and line 1730 will generate any patches from base_font to any of the next_font's.

Reply:
In that case what is the form or type of a "patch"? In (what was) 1730 a patch is a tuple that includes a `patch_url` but in this line neither `patch_url` nor `base_font` are passed to `encode`, so it doesn't seem likely it would be returning a tuple that includes that value.

Reply:
Oh sorry for the confusion, I forgot to rename "encode(" to "encode_fonts(" (just pushed a fix). This is meant to be a recursive call so the return is list of tuples of the (binary patch, url). In the base case where encode_fonts(...) is generating the patches to the leaf nodes line 1730 will create the initial set of tuples, and then each recursive call above will add additional tuples for it's specific layer in the graph.